test/test-notification-model-updates.js

```diff
@@ -0,0 +1,382 @@
+    var q = {where: {
+      workerId: 0,
+      serviceInstanceId: 1
+    }};
+    app.models.ServiceProcess.find(q, function(err, procs) {
+      tt.ok(!err, 'process should be found');
+      tt.equal(procs.length, 1, 'only one process should be found');
+      var proc = procs[0] || {};
+      tt.equal(proc.parentPid, 3456, 'parent pid should match');
+      tt.equal(proc.pid, 1234, 'pid should match');
+      tt.ok(!proc.isTrackingObjects, 'tracking-obj should be false');
+      tt.ok(!proc.isProfiling, 'tracking-obj should be false');
+      tt.ok(!proc.isSnapshotting, 'tracking-obj should be false');
+      tt.ok(!!proc.startTime, 'start time should be set');
+      tt.ok(!proc.stopTime, 'stop time should not be set');
+      tt.ok(!proc.stopReason, 'stop reason should not be set');
+      tt.end();
+    });
+  });
+
+  t.test('Notify fork', function(tt) {
+    var notification = {
+      cmd: 'fork',
+      id: 1,
+      pid: 1235,
+      ppid: 1234,
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  });
+
+  t.test('Notify listen', function(tt) {
+    var notification = {
+      cmd: 'listening',
+      id: 1,
+      pid: 1235,
+      address: {
+        address: '0.0.0.0',
+        port: 59176,
+        addressType: 4
+      }
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  });
+
+  function checkWorker1(prof, tt) {
+    var q = {where: {
+      workerId: 1,
+      serviceInstanceId: 1
+    }};
+    app.models.ServiceProcess.find(q, function(err, procs) {
+      tt.ok(!err, 'process should be found');
+      tt.equal(procs.length, 1, 'only one process should be found');
+      var proc = procs[0] || {};
+      tt.equal(proc.parentPid, 1234, 'parent pid should match');
+      tt.equal(proc.pid, 1235, 'pid should match');
+      tt.equal(proc.isTrackingObjects, prof.isTrackingObjects || false,
+        'tracking-obj should match');
+      tt.equal(proc.isProfiling, prof.isProfiling || false,
+        'cpu-profiling should match');
+      tt.equal(proc.isSnapshotting, prof.isSnapshotting || false,
+        'heap-snapshotting should match');
+      tt.ok(!!proc.startTime, 'start time should be set');
+      tt.ok(!proc.stopTime, 'stop time should not be set');
+      tt.ok(!proc.stopReason, 'stop reason should not be set');
+      tt.deepEqual(proc.listeningSockets, [
+        {
+          address: '0.0.0.0',
+          port: 59176,
+          addressType: 4
+        }
+      ]);
+      tt.end();
+    });
+  }
+
+  function notifyProfiling(type, value, tt) {
+    var notification = {
+      cmd: type,
+      id: 1,
+      pid: 1235,
+      isRunning: value
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  }
+
+  t.test('Notify object-tracking on',
+    notifyProfiling.bind(this, 'object-tracking', true));
```

Comment:
Could import it as `var partial = require('lodash').partial;` to make it shorter if you like.

For me the readability improvement comes from seeing right away that it is a partial application rather than some sort of instance method that returns a function.

But I don't care enough to make you change it, that's just my reasoning.